### PR TITLE
Hits and misses are now colored and display after each turn 

### PR DESCRIPTION
**Pull request by TimBeckmann**

### CODE REVIEW

**Comment by TimBeckmann** [comment]
The coloring might not work with other kind of terminal

**Comment by TimBeckmann** [comment]
The changes on this branch are based on the changes from PR#51

File: src/output.cpp [hunk 1]
```diff
@@ -67,7 +67,12 @@ void printGameBoard(
             const auto fieldValue = board[lineIndex][columnIndex];
             std::cout << paddingStr;
             try {
-                std::cout << translationTable.at(fieldValue);
+                if (fieldValue == FieldValue::SHIP_HIT)
+                    std::cout << "\033[1;31mX\033[0m";
+                else if (fieldValue == FieldValue::MISS)
+                    std::cout << "\033[1;34m~\033[0m";
+                else
+                    std::cout << translationTable.at(fieldValue);
```
Comment (jotoho) [inline src/output.cpp:75]: This shouldn't be hardcoded.

**Review by jotoho — CHANGES_REQUESTED** [review]

File: src/gamelogic.cpp [hunk 2]
```diff
@@ -150,7 +150,7 @@ WinnerID playGame(BoardType& playerBoard, BoardType& computerBoard) {
                 return potentialWinnerID;
         }
 
-        if (!playerHitOnce) {
+        if (!playerHit) {
```
Comment (jotoho) [inline src/gamelogic.cpp:153]: Revert the changes to the while loop and simply get rid of the if condition for the code under it

**Comment by TimBeckmann** [comment]
https://go.codetogether.com/#/26531c0c-39b7-4f4a-ad75-295f23fc8e9d/wQ76f058EVucUkfyvOzXth

File: src/output.cpp (hunk 1 above)
Comment (jotoho) [reply src/output.cpp]: Resolved

**Review by jotoho — APPROVED** [review]
